### PR TITLE
fix(color): Improvements to NV14/EL18 dead zone logic.

### LINKED ISSUE #2158: Settings do not get saved over a reboot (Internal TX setting, dead zone setting) on 2.7.1 on a NV14
### Is there an existing issue for this problem?

- [X] I have searched the existing issues
- It is vaguely similar to #2130 but in my case just the settings from the hardware tab get reset to default - no models are deleted.

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

I usually set the internal TX function to off while using the simulator and also I set the deadzone to 0 (from 4).
These settings get reverted (TX to ON, deadzone to 4) on reboot / restart of the Nirvana NV14 on EdgeTX 2.7.1

### Expected Behavior

I would expect those settings to remain saved over a reboot / restart.

### Steps To Reproduce

1. Go to to: Radio Settings --> Hardware tab
2. Set "Internal RF / internal module" to OFF, set deadzone (under 'sticks') to 0
3. Switch the radio off and on again.
4. Go back to the Hardware tab, settings are reverted to their default states.

### Version

2.7.1

### Transmitter

FlySky NV14

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by philmoz**
Fixes #2158

### CODE REVIEW

**Comment by philmoz** [comment]
Fixes:
- Don't overwrite NV14/EL18 dead zone when loading radio settings.
- Refactor dead zone code to remove floating point calculations.

File: radio/src/storage/storage_common.cpp [hunk 1]
```diff
@@ -100,18 +100,6 @@ void postRadioSettingsLoad()
...
-    g_eeGeneral.stickDeadZone = DEFAULT_STICK_DEADZONE;
-  }
-#endif
-#if !defined(DEBUG)
-  // clean up leftovers from a previous DEBUG config
-  for (uint8_t port_nr = 0; port_nr < MAX_AUX_SERIAL; port_nr++) {
-    if (serialGetMode(port_nr) == UART_MODE_DEBUG)
-      serialSetMode(port_nr, UART_MODE_NONE);
-  }
-#endif
```
Comment (pfeerick) [inline radio/src/storage/storage_common.cpp:114]: I think you accidentally included this on the rebase

Comment (philmoz) [reply radio/src/storage/storage_common.cpp]: Thanks. I've put the code back in.

**Comment by pfeerick** [comment]
> I would suggest simply removing the code that overwrites the dead zone in the postRadioSettingsLoad function.
The dead zone is set to the default value when the radio.yml file is first created (generalDefaults function) - if the user sets the value to 0 we should not be updating it after loading the radio.yml settings.

I believe the underlying issue there was to do with conversions, from when that capability didn't exist. As the `generalDefaults` function does not apply in this instance. 

This is also why the lines directly above exist
```
 #if defined(DEFAULT_INTERNAL_MODULE)
  if (!g_eeGeneral.internalModule) {
    g_eeGeneral.internalModule = DEFAULT_INTERNAL_MODULE;
  }
#endif
```

However, we don't seem to be able to tell the difference between the value not having been set at all vs having been set by the user (which I presume applies to both cases). But I think enough time has gone by though, and there is less chance of conversions from older firmware now.

**Pull request merged**